com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -334,12 +343,18 @@ public void NetworkHide(ulong clientId)
             // Send destroy call
             Observers.Remove(clientId);
 
-            using (var buffer = PooledNetworkBuffer.Get())
-            using (var writer = PooledNetworkWriter.Get(buffer))
+            var context = NetworkManager.MessageQueueContainer.EnterInternalCommandContext(
+                MessageQueueContainer.MessageType.DestroyObject,
+                NetworkChannel.Internal,
+                new[] {clientId},
+                NetworkUpdateLoop.UpdateStage
+            );
+            if (context != null)
             {
-                writer.WriteUInt64Packed(NetworkObjectId);
-
-                NetworkManager.MessageSender.Send(clientId, NetworkConstants.DESTROY_OBJECT, NetworkChannel.Internal, buffer);
+                using (var icontext = (InternalCommandContext) context)
```

Comment:
`icontext`? :D maybe `icmdctx` would be a less ambigious name :)

Reply:
why not just call it `internalCommandContext`?

Reply:
Really, I just needed to assign the `context` variable to a different name so I could cast it to not-null. I don't know why I chose the `i` prefix. I think `concreteContext` or `notNullContext` or maybe `reallyRealContextAndDefinitelyNotNull_IPromise` would be more accurate for why I'm reassigning it.